src/lib/server/events/gcal.ts

```diff
@@ -41,139 +35,99 @@ export function parse(rawICAL: string, options?: ICALParseOptions): ClubEvent[]
   return sortedEvents;
 }
 
-export interface ICALResolvable {
-  [k: string | number]: ICALResolvable | ICALResolvable[] | string;
-}
-
-/**
- * The code in this function is derived from
- * <https://github.com/adrianlee44/ical2json>.
- * @param rawICAL The raw calendar data in ICAL format.
- * @returns The parsed ICAL data.
- */
-export function* walkICAL(rawICAL: string) {
-  const output: ICALResolvable = {};
-  const lines = rawICAL.split(/\r\n|\n|\r/);
-  const parents: ICALResolvable[] = [];
-  let parent: ICALResolvable = {};
-  let current = output;
-  let currentKey = '';
-
-  for (const line of lines) {
-    let currentValue = '';
-    if (line.charAt(0) === ' ') {
-      current[currentKey] += line.substring(1);
-    } else {
-      const splitAt = line.indexOf(':');
-      if (splitAt < 0) {
-        continue;
-      }
-      currentKey = ((key: string) => {
-        if (key.startsWith('DTSTART')) return 'DTSTART';
-        if (key.startsWith('DTEND')) return 'DTEND';
-        return key;
-      })(line.substring(0, splitAt));
-      currentValue = line.substring(splitAt + 1);
-      switch (currentKey) {
-        case 'BEGIN': {
-          parents.push(parent);
-          parent = current;
-          if (parent[currentValue] == null) {
-            parent[currentValue] = [];
-          }
-          // Create a new object, store the reference for future uses.
-          current = {};
-          (parent[currentValue] as ICALResolvable[]).push(current);
-          break;
-        }
-        case 'END': {
-          current = parent;
-          parent = parents.pop() as ICALResolvable;
-          break;
-        }
-        default: {
-          if (current[currentKey]) {
-            if (!Array.isArray(current[currentKey])) {
-              current[currentKey] = [current[currentKey]] as ICALResolvable[];
-            }
-            (current[currentKey] as ICALResolvable[]).push(
-              currentValue as unknown as ICALResolvable
-            );
-          } else {
-            (current[currentKey] as string) = currentValue;
-          }
-        }
-      }
-    }
-  }
-
-  for (const icalEvent of (output['VCALENDAR'][0] as ICALResolvable)[
-    'VEVENT'
-  ] as ICALResolvable[]) {
-    yield icalEvent;
-  }
-}
-
-export function parseRRULE(rawRRULE: string): boolean {
-  if (rawRRULE === undefined) return false;
-
-  try {
-    const recurrence = RRule.fromString(rawRRULE);
-    return recurrence.isFullyConvertibleToText();
-  } catch {
-    return false;
+function makeClubEvent(event: GCalEvent, refDate: Temporal.ZonedDateTime): ClubEvent | null {
```

Comment:
Move `makeClubEvent` and helpers to `src/lib/server/events/event.ts`.
Note: Move `makeClubEvent` tests to `src/lib/server/events/event.test.ts`.